cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/workflow/WorkflowNodeCreator.java

```diff
@@ -51,9 +52,16 @@ static WorkflowNode createWorkflowActionNode(String programName, SchedulableProg
     return new WorkflowActionNode(programName, new ScheduleProgramInfo(programType, programName));
   }
 
+  @Deprecated
   static WorkflowNode createWorkflowCustomActionNode(WorkflowAction action) {
     Preconditions.checkArgument(action != null, "WorkflowAction is null.");
     WorkflowActionSpecification spec = DefaultWorkflowActionConfigurer.configureAction(action);
     return new WorkflowActionNode(spec.getName(), spec);
   }
+
+  static WorkflowNode createWorkflowCustomActionNode(CustomAction action) {
+    Preconditions.checkArgument(action != null, "CustomAction is null.");
+    WorkflowActionSpecification spec = DefaultWorkflowActionConfigurer.configureAction(action);
```

Comment:
this doesn't seem like it would compile or am I missing something? I don't see a method in DefaultWorkflowActionConfigurer that takes a CustomAction, and CustomAction doesn't extend WorkflowAction